codegen/src/test/resources/software/amazon/awssdk/codegen/poet/client/test-endpoint-discovery-async.java

```diff
@@ -255,9 +302,19 @@ public final String serviceName() {
                             .withOperationName("TestDiscoveryRequired")
                             .withMarshaller(new TestDiscoveryRequiredRequestMarshaller(protocolFactory))
                             .withResponseHandler(responseHandler).withErrorResponseHandler(errorResponseHandler)
-                            .discoveredEndpoint(cachedEndpoint).withInput(testDiscoveryRequiredRequest));
+                            .withMetricCollector(apiCallMetricCollector).discoveredEndpoint(cachedEndpoint)
+                            .withInput(testDiscoveryRequiredRequest));
+            AwsRequestOverrideConfiguration requestOverrideConfig = testDiscoveryRequiredRequest.overrideConfiguration().orElse(
+                    null);
+            executeFuture.whenComplete((r, e) -> {
+                List<MetricPublisher> metricPublishers = resolveMetricPublishers(clientConfiguration, requestOverrideConfig);
+                metricPublishers.forEach(p -> p.publish(apiCallMetricCollector.collect()));
+            });
             return executeFuture;
```

Comment:
Should we return whenComplete's result, so that it's easier to say that "when the execute future completes, metrics have been published"?

Reply:
+1